certbot-apache/certbot_apache/interfaces.py

```diff
@@ -110,35 +113,30 @@ class ParserNode(object):
     it's responsibility of each ParserNode object itself to store its prepending
     whitespaces in order to be able to reconstruct the complete configuration file
     as it was when originally read from the disk.
-    """
 
-    @property
-    @abc.abstractmethod
-    def ancestor(self):  # pragma: no cover
-        """
-        This property contains a reference to ancestor node, or None if the node
-        is the root node of the configuration tree.
+    ParserNode objects should have the following attributes:
 
-        :returns: The ancestor BlockNode object, or None for root node.
-        :rtype: ParserNode
-        """
+    # Reference to ancestor node, or None if the node is the root node of the
+    # configuration tree.
+    ancestor: Optional[ParserNode]
+
+    # True if this node has been modified since last save.
+    dirty: bool
 
-        raise NotImplementedError
+    # Filepath of the file where the configuration element for this ParserNode
+    # object resides.
+    filepath: str
+    """
 
-    @property
     @abc.abstractmethod
-    def dirty(self):  # pragma: no cover
+    def metadata(self, key):
```

Comment:
Just to make sure we're on the same page, you're planning on deleting this method based on https://github.com/certbot/certbot/pull/7308#discussion_r314388990 right?

Reply:
That's right!